_posts/2016-08-18-emerging-docker.md

```diff
@@ -0,0 +1,16 @@
+---
+layout: post
+title: Running Docker Natively on Windows and Mac
+---
+
+[Docker](https://www.docker.com) is an emerging technology for packaging software in so-called containers that can run processes isolated from the rest of the operating system. It is the midway point between a complete virtual machine for a single system and running a process with limited user rights.
+
+Docker provides so-called software images that package software. These images can then be executed inside a docker container. Furthermore, Docker provides a service called [DockerHub](https://hub.docker.com/) that allows users to upload and publish their own images. This allows developers to bundle their software and distribute it easily. Furthermore, since each container runs isolated from the rest of the systems, the developers do not have to rely on any kind of other system configuration.
+
+Within the OpenDreamKit project this allows us to bundle mathematical software systems (such as [GAP](http://www.gap-system.org/), [Sage](http://www.sagemath.org/), etc) and distribute them in a manner that is both accessible to users and easily maintainable for developers. Furthermore the isolated aspect of docker containers allows us to easily integrate multiple systems together without having to make additional assumptions about the user's specific setup -- we can just run all systems in one docker container. This way users can install the entire mathematical software stack that the OpenDreamKit project aims to provide easily.
+
+Docker was originally a Linux only application -- it relied on a lot of functionality provided by the Linux kernel. To make it available on Windows and Mac the developers provide a virtual machine, called [Docker Toolbox](https://www.docker.com/products/docker-toolbox), that runs a streamlined Linux system with Docker pre-installed. This makes it possible for Windows and Mac users to run docker containers,  however it introduces an additional layer of abstraction that comes with some disadvantages. The additional virtualisation slows down docker containers and faces technical limitations when wanting to integrate with the host system. It also requires users to install virtualisation software ([VirtualBox](https://www.virtualbox.org/) is installed by Docker Toolbox) before being able to run any kind of Docker Image. Even though it was automatically installed, this was a seperate application that was installed on users machines.
```

Comment:
I had moved all this paragraph to the present tense because Docker Toolbox is still distributed. However, the last sentence you just added is in past tense.